src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -380,12 +377,11 @@ export async function getMatchedDeposit(
 export function isUBAActivatedAtBlock(hubPoolClient: HubPoolClient, block: number, chain: number): boolean {
   try {
     const ubaActivationBlocks = getUbaActivationBundleStartBlocks(hubPoolClient);
-    const ubaActivationStartBlockForChain = getBlockForChain(
-      ubaActivationBlocks,
-      chain,
-      hubPoolClient.configStoreClient.enabledChainIds
-    );
-    return block >= ubaActivationStartBlockForChain;
+    const enabledChainsIndices = hubPoolClient.configStoreClient.getEnabledChains();
+    // Find the first activation block where the index matches the chain
+    const activationBlock =
```

Comment:
I'm not sure this is correct, can we just use `getBlockForChain` using `enabledChainsIndices`?

Reply:
Actually this should be fine